datatypes/src/collections/batch_builder.rs

```diff
@@ -0,0 +1,443 @@
+        }
+    }
+
+    pub fn points(
+        types: HashMap<String, FeatureDataType>,
+        num_features: usize,
+        num_coords: usize,
+    ) -> Self {
+        Self {
+            types,
+            column_arrays: HashMap::new(),
+            time_array: None,
+            geo_array: None,
+            num_features,
+            num_coords,
+            num_lines: None,
+            num_polygons: None,
+            num_rings: None,
+            output: None,
+            output_type: VectorDataType::MultiPoint,
+        }
+    }
+
+    pub fn lines(
+        types: HashMap<String, FeatureDataType>,
+        num_features: usize,
+        num_lines: usize,
+        num_coords: usize,
+    ) -> Self {
+        Self {
+            types,
+            column_arrays: HashMap::new(),
+            time_array: None,
+            geo_array: None,
+            num_features,
+            num_coords,
+            num_lines: Some(num_lines),
+            num_polygons: None,
+            num_rings: None,
+            output: None,
+            output_type: VectorDataType::MultiLineString,
+        }
+    }
+
+    pub fn num_features(&self) -> usize {
+        self.num_features
+    }
+
+    pub fn num_coords(&self) -> usize {
+        self.num_coords
+    }
+
+    pub fn num_lines(&self) -> Result<usize> {
+        self.num_lines
+            .ok_or_else(|| FeatureCollectionError::WrongDataType.into())
+    }
+
+    pub fn num_polygons(&self) -> Result<usize> {
+        self.num_polygons
+            .ok_or_else(|| FeatureCollectionError::WrongDataType.into())
+    }
+
+    pub fn num_rings(&self) -> Result<usize> {
+        self.num_rings
+            .ok_or_else(|| FeatureCollectionError::WrongDataType.into())
+    }
+
+    pub fn column_types(&self) -> HashMap<String, FeatureDataType> {
+        self.types.clone()
+    }
+
+    pub fn set_time_intervals(&mut self, values_buffer: Buffer) -> Result<()> {
+        let data = ArrayData::builder(TimeInterval::arrow_data_type())
+            .len(self.num_features)
+            .add_buffer(values_buffer)
+            .build();
+
+        let array = Arc::new(FixedSizeListArray::from(data)) as ArrayRef;
+
+        self.time_array = Some(array);
+
+        Ok(())
+    }
+
+    pub fn set_default_time_intervals(&mut self) -> Result<()> {
+        let mut time_intervals_builder = TimeInterval::arrow_builder(self.num_features);
+
+        let default = TimeInterval::default();
+        for _ in 0..self.num_features {
+            let date_builder = time_intervals_builder.values();
+            date_builder.append_value(default.start().inner())?;
+            date_builder.append_value(default.end().inner())?;
+
+            time_intervals_builder.append(true)?;
+        }
+
+        let array = Arc::new(time_intervals_builder.finish()) as ArrayRef;
+
+        self.time_array = Some(array);
+
+        Ok(())
+    }
+
+    pub fn set_points(&mut self, coords: Buffer, offsets: Buffer) -> Result<()> {
```

Comment:
I'm not quite sure if our interface should use the arrow types. If the builder is only meant to be used for OpenCL, then it's fine, but otherwise we need some wrapper type I guess.

Reply:
maybe we can call it ArrowBufferBuilder or something like that. It's not necessarily only meant for OpenCl, but for cases where the arrow buffers are built externally and we want to avoid copies. Maybe we can also use this for Python integration or results computed on a different machine.

I don't think wrapping the buffers makes any sense because nothing of it works without arrow ;)

Reply:
This is valid. It is like an builder from internals, raw parts, or something like that.

Reply:
ec9ee747f1e86433c9466fe506e5ddf216651827